src/msg/cli.rs

```diff
@@ -258,308 +341,327 @@ fn msg_matches_search(ctx: &Ctx, matches: &clap::ArgMatches) -> Result<bool> {
+    // ------------
     imap_conn.logout();
     Ok(true)
 }
 
 fn msg_matches_move(ctx: &Ctx, matches: &clap::ArgMatches) -> Result<bool> {
-    debug!("move command matched");
+    debug!("Move command matched");
 
+    // -----------------
+    // Preparations
+    // -----------------
+    let mut imap_conn = ImapConnector::new(&ctx.account)?;
     let uid = matches.value_of("uid").unwrap();
-    debug!("uid: {}", &uid);
     let target = matches.value_of("target").unwrap();
-    debug!("target: {}", &target);
+    let mut msg = imap_conn.get_msg(&ctx.mbox, &uid)?;
 
-    let mut imap_conn = ImapConnector::new(&ctx.account)?;
-    let msg = Msg::from(imap_conn.read_msg(&ctx.mbox, &uid)?);
-    let mut flags = msg.flags.to_vec();
-    flags.push(Flag::Seen);
-    imap_conn.append_msg(target, &msg.raw, flags)?;
-    imap_conn.add_flags(&ctx.mbox, uid, "\\Seen \\Deleted")?;
-    debug!("message {} successfully moved to folder `{}`", uid, target);
+    debug!("Uid: {}", &uid);
+    debug!("Target: {}", &target);
+
+    // -----------
+    // Action
+    // -----------
+    // create the mail in the target-mailbox
+    msg.flags.insert(Flag::Seen);
+
+    imap_conn.append_msg(target, &mut msg)?;
+
+    debug!("Message {} successfully moved to folder `{}`", uid, target);
     ctx.output.print(format!(
         "Message {} successfully moved to folder `{}`",
         uid, target
     ));
 
+    // mark the current mail in the current mailbox as deleted
+    imap_conn.add_flags(&ctx.mbox, uid, "\\Seen \\Deleted")?;
+
+    // remove the current mail from the current mailbox
     imap_conn.expunge(&ctx.mbox)?;
+
+    // be nice to the server and say "bye" ;)
     imap_conn.logout();
     Ok(true)
 }
 
 fn msg_matches_delete(ctx: &Ctx, matches: &clap::ArgMatches) -> Result<bool> {
-    debug!("delete command matched");
+    debug!("Delete command matched");
 
+    // -----------------
+    // Preparations
+    // -----------------
+    let mut imap_conn = ImapConnector::new(&ctx.account)?;
     let uid = matches.value_of("uid").unwrap();
-    debug!("uid: {}", &uid);
 
-    let mut imap_conn = ImapConnector::new(&ctx.account)?;
+    debug!("Uid: {}", &uid);
+
+    // -----------
+    // Delete
+    // -----------
     imap_conn.add_flags(&ctx.mbox, uid, "\\Seen \\Deleted")?;
-    debug!("message {} successfully deleted", uid);
+    imap_conn.expunge(&ctx.mbox)?;
+
+    debug!("Message {} successfully deleted", uid);
     ctx.output
         .print(format!("Message {} successfully deleted", uid));
 
-    imap_conn.expunge(&ctx.mbox)?;
+    // ------------
+    // Cleanup
+    // ------------
     imap_conn.logout();
     Ok(true)
 }
 
 fn msg_matches_send(ctx: &Ctx, matches: &clap::ArgMatches) -> Result<bool> {
-    debug!("send command matched");
+    debug!("Send command matched");
 
     let mut imap_conn = ImapConnector::new(&ctx.account)?;
 
-    let msg = if atty::is(Stream::Stdin) {
+    let msg: String = if atty::is(Stream::Stdin) {
```

Comment:
I don't think it is necessary to have the type here and with the collect

Reply:
When I saw it for the first time, I didn't knew what I got. So it helped for me to know the data type here. But if you don't want to have it, then I can remove it. (But I'd like to let it like that).